pkg/cloudlogging/puller.go

```diff
@@ -111,3 +113,34 @@ func (p *Puller) Pull(ctx context.Context, filter string, maxCount int) ([]*logg
 	}
 	return ls, nil
 }
+
+// SteamPull pulls live log entries, it won't stop until a cancel signal happens.
+func (p *Puller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry, tailLogEntriesClient loggingpb.LoggingServiceV2_TailLogEntriesClient) (chan<- *loggingpb.LogEntry, error) {
```

Comment:
The func shouldn't accept the stream client. It should be invisible to the caller.

Reply:
I thought the caller should provide the client, so it can close the client by itself, therefore StreamPull don't need to handle the close of the client?

Reply:
The stream client should be generated by calling logging client in this func. With that, it's OK to close the stream client here (you create it and close it in the same scope).

Reply:
How about making the Puller struct to have a stream client.

Reply:
Updated to create and close stream client in streamPull